src/codegen/generators/variable_generator.rs

```diff
@@ -115,8 +115,13 @@ impl<'ctx, 'b> VariableGenerator<'ctx, 'b> {
         let type_name = global_variable.get_type_name();
         let variable_type = self.types_index.get_associated_type(type_name)?;
 
-        let mut global_ir_variable =
-            self.llvm.create_global_variable(self.module, global_variable.get_name(), variable_type);
+        let name = if self.global_index.get_type_information_or_void(type_name).is_enum() {
+            global_variable.get_qualified_name()
+        } else {
+            global_variable.get_name()
```

Comment:
Anything against just using the get_quailified_name here? I think for now it should deliver the same for global variables

Reply:
Good point, I'll give it a try

Reply:
This approach will break tests/code depending on `inkwell::add_global_function_mapping`. Subsequent calls to `ExecutionEngine::get_function` will then be unable to find the given functions by name in `GeneratedModule::run` and cause tests to `panic` on `unwrap`.

Somehow this seems to affect PROGRAMs but not FUNCTION_BLOCKs or FUNCTIONs, I guess since PROGRAMs are global by default?

```
PROGRAM foo
END_PROGRAM

FUNCTION_BLOCK bar  
END_FUNCTION_BLOCK

FUNCTION baz
END_FUNCTION
```

```
%foo = type {}
%bar = type {}

@foo = global %foo zeroinitializer
@__bar__init = unnamed_addr constant %bar zeroinitializer

define void @foo.1(%foo* %0) {
entry:
  ret void
}

define void @bar(%bar* %0) {
entry:
  ret void
}

define void @baz() {
entry:
  ret void
}
```
`foo` is now being declared as `foo.1` - that's quite an interesting side-effect.
I can try filtering programs instead of enums and see if that fixes it/if there are then other side-effects

Reply:
yeah foo.1 would not be good as it probably won't be a reliable result for the test, also that meens calling the function from C will have issue i assume. Shouldn't the global initializer for foo be called something else? we usually called it foo_instance or __foo__init or something similar. This is probably what qualified name is not doing.. interesting

Reply:
Filtering out programs instead of enums seems promising, but I need to write additional tests. These would involve linking against a precompiled library which exposes the globals and then check for expected parameters. I got sidetracked while working on another issue, but I'll try to get on this tomorrow, so we can merge/close this.

Reply:
Let's pair on this before you change the behavior / filter out programs.